lib/cn/kvcompact.c

```diff
@@ -0,0 +1,452 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
+ */
```

Comment:
Putting this in its own file duplicates 200 lines of non-trivial code, including the following functions:
- merge_item_compare()
- replenish()
- merge_init()
- get_next_item()
- get_kvset_builder()
- get_direct_read_buf()

And only adds one new function:  `kv_compact()`.

Why not add `kv_compact()` to end of `spill.c` and eliminate all the duplication?

Reply:
Funny, I think that's where he originally had it!

Reply:
I think a common.h/.c approach is better here, and have all the cn actions like spill, kcompact, kvcompact, split, as their own source files.

Reply:
If you factor the common code out, then you lose the private namespace of having it in one file.  Since kvcompact is only one function, I think it makes sense to stash in `spill.c`.  And maybe rename `spill.c` to `spill_kvcompact.c`?

Reply:
Also inhibits the compiler from inlining them, unless you make them all static inline functions...
Let's not rename spill.c in this patch please!